cloudify_hostpool/tests/config/test_yaml_pool.py

```diff
@@ -184,3 +185,36 @@ def test_empty_config(self):
             exceptions.ConfigurationError,
             'Pool configuration is missing a hosts section',
             yaml_pool.YAMLPoolLoader, {})
+
+    def test_bad_keyfile(self):
+        tmp_dir = tempfile.mkdtemp()
+        config = {'default': {'auth': {'username': 'x'}, 'port': 123},
+                  'hosts': [{'host': 'google.com',
+                             'auth': {'username': 'x',
+                                      'keyfile': tmp_dir + '/wrong_file'},
+                             'port': 80}]}
+        try:
+            pool = yaml_pool.YAMLPoolLoader(config)
+            self.assertRaisesRegexp(exceptions.ConfigurationError,
+                                    'There is no key file:', pool.load().next)
+        finally:
+            os.rmdir(tmp_dir)
+
+    def test_good_keyfile(self):
+        fd, good_file = tempfile.mkstemp()
+        os.close(fd)
```

Comment:
"_have to close_" = the function also returns an open file descriptor (because you usually want to use the temporary file you just have created).